### ISSUE TITLE
Duplicate AMI

### ISSUE THREAD

**Issue by waiteb3** [mitchellh/packer#2123]
When coping AMIs, it duplicates the AMI even if it was built in the same region

```json
{
  "region": "us-west-2",
  "ami_regions": [
    "us-east-1",
    "us-west-2"
  ]
}
```

It's only mildly annoying, but it'd be nice if it was either documented or dealt with in the logic.

**Comment by henrysher** [comment]
@mitchellh my PR failed in the ci test, however, it seems the failure was not caused by my code.

**Comment by mitchellh** [comment]
Fixed by #2175
